src/components/Profile/FollowButton.tsx

```
@@ -36,7 +39,7 @@ const FollowButton = memo(function FollowButton({ profile }: FollowButtonProps)
             disabled={loading}
             onMouseEnter={() => setFollowHover(true)}
             onMouseLeave={() => setFollowHover(false)}
-            onClick={handleToggle}
+            onClick={() => isLogin && handleToggle()}
```

It's better to draft the login modal better than failing silently.